src/components/expression/expression-renderer.js

```diff
@@ -452,6 +515,99 @@ genotet.ExpressionRenderer.prototype.drawMatrixConditionLabels_ = function() {
 };
 
 /**
+ * Renders the expression heatmap gradient.
+ * @private
+ */
+genotet.ExpressionRenderer.prototype.drawHeatmapGradient_ = function() {
+  this.svgHeatmapGradient_.selectAll('*').remove();
+  var heatmapData = this.data.matrix;
+  var scaleMin = 0;
+  var scaleMax = 0;
+  if (this.data.options.autoScaleGradient) {
+    scaleMin = heatmapData.valueMin;
+    scaleMax = heatmapData.valueMax;
+  }
+  else {
+    scaleMin = heatmapData.allValueMin;
+    scaleMax = heatmapData.allValueMax;
+  }
+  scaleMin = scaleMin.toFixed(2);
+  scaleMax = scaleMax.toFixed(2);
+  var marginLeft = scaleMin.toString().length *
+    this.CONDITION_LABEL_HEIGHT_FACTOR_;
+  var marginRight = scaleMax.toString().length *
+    this.CONDITION_LABEL_HEIGHT_FACTOR_;
+
+  var transformLeft = 0;
+  var transformTop = this.heatmapHeight_;
+  if (this.data.options.showProfiles) {
+    transformTop += this.profileHeight_;
+  }
+  if (this.data.options.showGeneLabels) {
+    transformLeft += this.geneLabelWidth_ + this.LABEL_MARGIN_;
+  }
+  else if (this.data.options.showProfiles) {
+    transformLeft += this.DEFAULT_PROFILE_MARGIN;
+  }
+  if (this.data.options.showConditionLabels) {
+    transformTop += this.conditionLabelHeight_ +
+      this.LABEL_MARGIN_ + this.LABEL_DIFFERENCE_;
+  }
+  this.svgHeatmapGradient_
+    .attr(
+      'transform',
+      genotet.utils.getTransform([transformLeft, transformTop])
+    );
+  var gradientHeight = this.DEFAULT_HEATMAP_GRADIENT_HEIGHT -
+    this.HEATMAP_GRADIENT_MARGINS_.TOP -
+    this.HEATMAP_GRADIENT_MARGINS_.BOTTOM;
+  var gradientWidth = this.heatmapWidth_ - marginLeft - marginRight;
+  var gradientContent = this.svgHeatmapGradient_.append('defs')
```

Comment:
I'd recommend writing this gradient as a template somewhere in the node's template. It's your call though.